streams/src/main/java/org/apache/kafka/streams/state/internals/WrappedStateStore.java

```diff
@@ -19,81 +19,66 @@
 import org.apache.kafka.streams.errors.InvalidStateStoreException;
 import org.apache.kafka.streams.processor.ProcessorContext;
 import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.state.TimestampedBytesStore;
 
 /**
  * A storage engine wrapper for utilities like logging, caching, and metering.
  */
-public interface WrappedStateStore extends StateStore {
-
-    /**
-     * Return the inner most storage engine
-     *
-     * @return wrapped inner storage engine
-     */
-    StateStore inner();
-
-    /**
-     * Return the state store this store directly wraps
-     * @return the state store this store directly wraps
-     */
-    StateStore wrappedStore();
-
-    abstract class AbstractStateStore implements WrappedStateStore {
-        final StateStore innerState;
-
-        protected AbstractStateStore(final StateStore inner) {
-            this.innerState = inner;
+public class WrappedStateStore<S extends StateStore> implements StateStore {
+    public static boolean isTimestamped(StateStore stateStore) {
```

Comment:
Moving this in here because it requires specific knowledge of the concrete store types. Especially checking and recursively unwrapping wrapped stores is a potentially brittle and dangerout operation that shouldn't be done scattered throughout the codebase.

Specifically, as you recurse down the wrapper references, you lose all knowledge of the wrapped type. We avoid a dangerous cast by answering a specific question (is there a timestamped byte store in my wrapped stack of stores?) and returning a boolean, rather than a reference to an unwrapped and casted store.